pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -493,15 +493,20 @@ func (nc *Controller) doNoScheduleTaintingPassWorker(i int, stopCh <-chan struct
 		select {
 		case <-stopCh:
 			return
-		case node := <-nc.nodeUpdateChannels[i]:
-			if err := nc.doNoScheduleTaintingPass(node); err != nil {
-				glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", node.Name, err)
+		case nodeName := <-nc.nodeUpdateChannels[i]:
+			if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
+				glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", nodeName, err)
 			}
 		}
 	}
 }
 
-func (nc *Controller) doNoScheduleTaintingPass(node *v1.Node) error {
+func (nc *Controller) doNoScheduleTaintingPass(nodeName string) error {
+	node, err := nc.nodeLister.Get(nodeName)
```

Comment:
just to double check, doNoScheduleTaintingPass doesn't mutate anything on the node object, right?

Reply:
yes, we only need to get node's condition and taints, and set update request if condition/taint did not match.